### DISCUSSION TITLE
Client side navigation not working even on a brand new empty barebones project (App router)

### DISCUSSION THREAD

**Discussion by romiem** [vercel/next.js#62682] [Help] [answered]
### Summary

Since yesterday, client side navigation for all projects under my account have stopped working (for new commits) when deployed to Vercel. **It works fine on the local dev server. It also works fine if running as a production optimised build locally (```npm run build``` and then ```npm run start```)**.

I've created a sample barebones project below to demonstrate the issue on Vercel. If you click the 'add background colour' button, the background colour is lost between page navigations (on dev the page colour is preserved). You can also observe the behaviour by looking at the network tab in chrome inspector.

This issue has affected all projects on my account. I don't know if this is particular to my account or a more general issue with client side routing configuration on Vercel. Given that both the dev and prod builds run fine locally, I have my suspicious there is a configuration issue for my apps on Vercel servers.

**Steps to Reproduce**
Open the link and click on the add background color button. Then navigate between routes. If client side navigation was occuring, the background color should remain yellow (works fine when running locally on the dev server). On production however, you can see that a full page refresh occurs and the colour is cleared (you can also see this with dev console open and in the network tab, you can see a full page refresh is occurring when navigating between routes).

**Git repo here:**
https://github.com/romiem/nextjs-nav

### Example

https://nextjs-nav-sable.vercel.app/

**Comment by styfle** [comment, marked as answer]
Thanks for providing the steps to reproduce! 

The issue has been resolved, thanks!

**Reply by clementohNZ** [reply]
Hi @styfle, just wondering how the issue was resolved? I have the same issue on 14.1.4

**Reply by styfle** [reply]
@clementohNZ It was a bug with some Vercel deployments. 

You can see the example above is working now https://nextjs-nav-sable.vercel.app

If you still have the issue please provide the steps to reproduce, thanks!

**Reply by clementohNZ** [reply]
Interesting... is it possible to know what the bug is?

I redeployed the application [www.lube.gg](https://www.lube.gg/), but the bug is still present. If you navigate between the register and login pages, you will see that it doesn't use client-side navigation.

The client-side navigation works locally no issues.

**Reply by styfle** [reply]
It looks like its because you are linking to a different domain. 

Try `<Link href="/login">` instead of `<Link href="https://lube.gg/login">` since its redirecting to www which you likely configured on your Vercel project.

<img width="481" alt="image" src="https://github.com/vercel/next.js/assets/229881/5533acc5-01ec-4519-9842-1318fb882969">

**Reply by clementohNZ** [reply]
Thank you so much @styfle! Turns out some changes to a custom router was converting the hrefs into an absolute path. Really appreciate your help digging into this.
